File: app/src/main/java/com/wangzhen/reader/ui/base/BaseActivity.java
package com.wangzhen.reader.ui.base;

import android.os.Bundle;

import androidx.annotation.LayoutRes;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.Toolbar;

import com.wangzhen.commons.toolbar.ToolbarActivity;

import butterknife.ButterKnife;
import butterknife.Unbinder;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

/**
 * Created by PC on 2016/9/8.
 */
public abstract class BaseActivity extends ToolbarActivity {
    protected CompositeDisposable mDisposable;
    private Unbinder unbinder;

    /****************************abstract area*************************************/

    @LayoutRes
    protected abstract int getContentId();

    /************************init area************************************/
    protected void addDisposable(Disposable d) {
        if (mDisposable == null) {
            mDisposable = new CompositeDisposable();
        }
        mDisposable.add(d);
    }

    protected void initData(Bundle savedInstanceState) {
    }

    /**
     * 初始化零件
     */
    protected void initWidget() {

    }

    /**
     * 初始化点击事件
     */
    protected void initClick() {
    }

    /**
     * 逻辑使用区
     */
    protected void processLogic() {
    }

    /*************************lifecycle area*****************************************************/

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getContentId());
        initData(savedInstanceState);
        unbinder = ButterKnife.bind(this);
        initWidget();
        initClick();
        processLogic();

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        unbinder.unbind();
        if (mDisposable != null) {
            mDisposable.dispose();
        }
    }
}

File: app/src/main/java/com/wangzhen/reader/ui/base/BaseActivity.kt
package com.wangzhen.reader.ui.base

import android.os.Bundle
import com.wangzhen.commons.toolbar.ToolbarActivity
import com.wangzhen.statusbar.DarkStatusBar
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable

/**
 * BaseActivity
 * Created by wangzhen on 2023/4/11
 */
open class BaseActivity : ToolbarActivity() {
    private var disposables = CompositeDisposable()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        fitDarkStatusBar()
    }

    protected fun addDisposable(d: Disposable) {
        disposables.add(d)
    }

    protected fun fitDarkStatusBar() {
        DarkStatusBar.get().fitDark(this)
    }

    protected fun fitLightStatusBar() {
        DarkStatusBar.get().fitLight(this)
    }

    override fun onDestroy() {
        super.onDestroy()
        disposables.dispose()
    }
}